internal/fs/fs.go

```diff
@@ -692,3 +692,50 @@ func volumeName(path string) (v string) {
 	}
 	return ""
 }
+
+// CopyTree recursively copies a source directory into a target directory.
+// Target directories are created if they don't exist yet.
+func CopyTree(dest, src string) error {
```

Comment:
We didn't have something like this already?

Reply:
We've got `CopyDir` - https://github.com/golang/dep/blob/22125cfaa6ddc71e145b1535d4b7ee9744fefff2/internal/fs/fs.go#L356

i think that should do it.

Reply:
I've added a separate commit which removes this code duplication.

This PR is starting to collect some commits which could also be merged separately, like that cleanup commit. Please let me know whether I should create separate PRs for that or whether you are fine with merging everything in one go.